distro_info.py

```diff
@@ -0,0 +1,171 @@
+import distro
+import locale
+import logging
+import platform
+import subprocess
+from collections import namedtuple
+from pathlib import Path
+
+try:
+	# distutils is part of the python standard library, but is deprecated,
+	# slated for removal in 3.12. setuptools has absorbed some (all) of
+	# distutils, so let's try their version first.
+	from setuptools._distutils.version import LooseVersion as DistroVersion
```

Comment:
There are the fallback cases, but otherwise, not really.
The idea is that we want a good version class without implementing it ourselves or pulling in a dependency outside of the ubiquitous python packages.
`distutils` used to be part of the standard library, and contains a suitable version class. `setuptools`, a ubiquitous python package, vendored the latest version of `distutils` before it was removed from the standard library, so we can keep using it from there. I imagine this will remain relatively stable until a more "proper" solution is available.

Reply:
That's fair.